media.go

```diff
@@ -468,8 +505,9 @@ func (ms *mediaSource[T, U]) Stream(ctx context.Context, errHandlers ...ErrorHan
 		producerCond := sync.NewCond(condMu)
 		consumerCond := sync.NewCond(condMu.RLocker())
 
+		rootCancelCtxWithSpan := trace.NewContext(ms.rootCancelCtx, trace.FromContext(ctx))
```

Comment:
`trace.FromContext` may return `nil` if there's no span. Would that cause a panic in `trace.NewContext`?

Reply:
Calling `trace.NewContext` with a `nil` span would append a `nil` span to the context which would later be handled gracefully if it was ever accessed, but it would be unnecessary to leave in there. The new changes don't add the span if it's `nil`.